packages/techdocs-common/src/stages/publish/googleStorage.test.ts

```diff
@@ -55,24 +79,164 @@ beforeEach(async () => {
 });
 
 describe('GoogleGCSPublish', () => {
-  it('should publish a directory', async () => {
-    mockFs({
-      '/path/to/generatedDirectory': {
-        'index.html': '',
-        '404.html': '',
-        assets: {
-          'main.css': '',
+  describe('publish', () => {
+    beforeEach(() => {
+      const entity = createMockEntity();
+      const entityRootDir = getEntityRootDir(entity);
+
+      mockFs({
+        [entityRootDir]: {
+          'index.html': '',
+          '404.html': '',
+          assets: {
+            'main.css': '',
+          },
         },
-      },
+      });
+    });
+
+    afterEach(() => {
+      mockFs.restore();
+    });
+
+    it('should publish a directory', async () => {
+      const entity = createMockEntity();
+      const entityRootDir = getEntityRootDir(entity);
+
+      expect(
+        await publisher.publish({
+          entity,
+          directory: entityRootDir,
+        }),
+      ).toBeUndefined();
+      mockFs.restore();
+    });
+
+    it('should fail to publish a directory', async () => {
+      const wrongPathToGeneratedDirectory = path.join(
+        rootDir,
+        'wrong',
+        'path',
+        'to',
+        'generatedDirectory',
+      );
+
+      const entity = createMockEntity();
+
+      await expect(
+        publisher.publish({
+          entity,
+          directory: wrongPathToGeneratedDirectory,
+        }),
+      ).rejects.toThrowError();
+
+      await publisher
+        .publish({
+          entity,
+          directory: wrongPathToGeneratedDirectory,
+        })
+        .catch(error => {
+          expect(error.message).toEqual(
```

Comment:
You should make sure that these actually get executed. Either add an `expect.assertions(3)` at the top, or do

```ts
.then(
  () => fail(),
  error => {
    ...
```

Reply:
That's a great feedback, thank you!